pkg/curatedpackages/bundle.go

```diff
@@ -0,0 +1,117 @@
+package curatedpackages
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere-packages/pkg/testutil"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/executables"
+)
+
+const (
+	Cluster         = "cluster"
+	Registry        = "registry"
+	RegistryBaseRef = "public.ecr.aws/q0f6t3x4/eksa-package-bundles"
+)
+
+func GetLatestBundle(ctx context.Context, kubeConfig string, source string, kubeVersion string) (*api.PackageBundle, error) {
+	var (
+		packageBundle *api.PackageBundle
+		err           error
+	)
+
+	switch strings.ToLower(source) {
+	case Cluster:
+		packageBundle, err = getActiveBundleFromCluster(ctx, kubeConfig)
+	case Registry:
+		packageBundle, err = getLatestBundleFromRegistry(ctx, kubeVersion)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return packageBundle, nil
+}
+
+func createBundleManager(kubeVersion string) (manager bundle.Manager) {
```

Comment:
Idiomatic Go would say to name this function `newBundleManager`.

Reply:
Drop the return variable also, it can have unintended side effects.

Reply:
Looking at the function further `new` isn't necessarily appropriate here. Its not wrong, but neither is `create`. `new` is generally used in initializing constructors and this isn't strictly an initializing constructor. It _uses_ an initializing constructor, `NewBundleManager`, but doesn't perform the initialization itself.